spec/requests/api/autocomplete_controller_spec.rb

```diff
@@ -0,0 +1,70 @@
+require 'rails_helper'
+
+RSpec.describe 'Api::AutocompleteController', type: :request do
+
+  let(:group) { create(:group) }
+  let(:john) { create(:user, email: 'john@commons.org') }
+  let(:mark) { create(:user, email: 'mark@commons.org') }
+  let(:jack) { create(:user, email: 'jack@commons.org') }
+
+  describe 'unauthorized' do
+    describe 'GET /autocomplete/members' do
+      it 'redirects to login page' do
+        get api_autocomplete_members_path
+        expect(response).to redirect_to new_user_session_path
+      end
+    end
+  end
+
+  describe 'authorized' do
+    before { sign_in john }
+
+    describe 'GET /autocomplete/members' do
+      context 'with group id' do
+        context 'without query' do
+          it 'gets 200' do
+            get api_autocomplete_members_path(group_id: group.id)
+            expect(response).to have_http_status(200)
+          end
+
+          it 'returns only the users that do not belong in the group' do
+            group.add_user(john)
+            group.add_user(mark)
+            jack
```

Comment:
Personal Preference: I often get confused with the style of testing where `let`s are used to DRY up test code. In this case, a declaration like `jack` is difficult to parse on first blush.

I'd almost prefer to just call `jack = create(:user, email: 'jack@commons.org')` once in each test, rather than have Mystery Guests.

More on why [here][1].

[1]: https://robots.thoughtbot.com/lets-not